### ISSUE TITLE
Add last successful sign-in properties to Get-MgUser signInActivity

### ISSUE THREAD

**Issue by nielsvdc** [microsoftgraph/msgraph-sdk-powershell#2465]
**Is your feature request related to a problem? Please describe.**
When using Get-MgUser the last sign-in datetime are returned in the SignInActivity object even if the sign-in failed, for example because to the account is disabled. But I only want to know the last successful sign-in. There's no easy way to filter this, other than to filter the sign-in or audit logs for each of the 8000+ accounts we have. So the extra properties would be valuable.

**Describe the solution you'd like**
4 extra properties should be added to the SignInActivity output.
{
  "@odata.type": "#microsoft.graph.signInActivity",
  "lastSignInDateTime": "String (timestamp)",
  "lastSignInRequestId": "String",
  "lastNonInteractiveSignInDateTime": "String (timestamp)",
  "lastNonInteractiveSignInRequestId": "String",
  "lastSuccessfulSignInDateTime": "String (timestamp)",
  "lastSuccessfulSignInRequestId": "String",
  "lastSuccessfulNonInteractiveSignInDateTime": "String (timestamp)",
  "lastSuccessfulNonInteractiveSignInRequestId": "String"
}

**Describe alternatives you've considered**
Do a loop for each account to filter the sign-in logs to get the last successful sign-in.

**Additional context**
Please check this thread: https://learn.microsoft.com/en-us/answers/questions/1337000/msgraph-signinactivity-includes-failed-logins?page=1&orderby=helpful&comment=answer-1292871

**Comment by timayabi2020** [comment]
Hi @nielsvdc  thank you for bringing this issue to our attention. As we plan to have that data availed to you, will using the ``Invoke-MgGraphRequest`` help? 
i.e. ``Invoke-MgGraphRequest -OutputType HashTable -Uri "htps://graph.microsoft.com/v1.0/users<your id here>`$select-signInActivity"``

The above will give a result similar to the screenshot below.
<img width="1221" alt="image" src="https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/10947120/fb158bc2-158e-4d52-b74d-e2b1308d7a96">

You can also use PSObject or Json as a value to the ``-OutputType`` parameter.

**Comment by nielsvdc** [comment]
Hi @timayabi2020. The output of Invoke-MgGraphRequest is the same. It only give the last attempts and not returning if the sign-in attempt was successful or failed. So the Invoke-MgGraphRequest output might also need some modification.

_I'm using Get-MgUser because I'm also retrieving other user information._

**Comment by timayabi2020** [comment]
@nielsvdc First of all apologies for misunderstanding the issue.
 Unfortunately, this is a service issue which we are not best placed to give an answer or implement the enhancement. Kindly raise an issue here https://developer.microsoft.com/en-us/graph/support so that the API owner can respond to it.

**Comment by worldsdream** [comment]
Hi @nielsvdc have you seen the below:
https://www.alitajran.com/export-microsoft-365-inactive-users/

You can get the last successful sign-in as shown there from the Graph SDK.

Ensure you update to the latest version. Otherwise, it will not work!

I am sure you can all the other properties.

**Comment by nielsvdc** [comment]
@worldsdream Yes, I know. But we have more than 8K accounts, so it would be nice to get this information through the Get-MgUser signInActivity object, instead of having to loop 8000 times.

**Comment by jantari** [comment]
@nielsvdc I think what @worldsdream meant to show you is that this information is available in the SignInActivity object, but only when you query the Graph Beta API.

E.g. using:

```powershell
$entraInfo = Get-MgBetaUser -Filter "UserPrincipalName eq 'upn@domain.tld'" -Property Id, SignInActivity;
$entraInfo.SignInActivity | Format-List *
```

or, if you don't want to install the Graph Beta modules just for this one request, you can also use:

```powershell
$UPN = 'upn@domain.tld'
$GraphQueryParams = [URI]::EscapeUriString("?`$filter=UserPrincipalName eq '$UPN'&`$select=Id,SignInActivity")
Invoke-MgGraphRequest -Method 'Get' -Uri "https://graph.microsoft.com/beta/users$GraphQueryParams"
```
Either way gives you the `LastSuccessfulSignInDateTime` property you were looking for.

**Comment by listen42one** [comment]
Is there any way to exclude non-interactive sign-ins for LastSuccessfulSignInDateTime? I just want to know the last time someone successfully logged in with their username and password and am having a surprisingly difficult time getting this data through a script.
